pkg/controller/node/node_controller.go

```diff
@@ -531,73 +545,73 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 	portList, err := listPortsByAttachmentIds(nsxClients, attachment_ids)
 	if err != nil {
 		cachedNodeSet[nodeName] = &statusmanager.NodeStatus{
-			Address: nodeAddress,
+			Addresses: nodeAddresses,
 			Success: false,
 			Reason:  fmt.Sprintf("Error while achieving ports for node %s: %v", nodeName, err),
 		}
 		r.status.SetFromNodes(cachedNodeSet)
 		return reconcile.Result{}, err
 	}
-	lsp, err := filterPortByNodeAddress(nsxClients, portList, nodeAddress)
+	lsps, err := filterPortsByNodeAddresses(nsxClients, portList, nodeAddresses)
```

Comment:
Even node has many addresses including external IP and internal IP, I don't think we need get addresses from multiple LSPs. It should be simpler if we have the limitation that node only attaches one vNIC to LS.

Reply:
ditto